### PR TITLE
EL-711: Cost of sale bug

### PR DESCRIPTION
**Pull request by patrick-laa**
[Jira ticket](https://dsdmoj.atlassian.net/browse/EL-711)

## What changed and why

Show new deductions row for main properties when there are deductions
Also don't show cost of sale deductions for additional properties when there are none (e.g. for controlled work cases)

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing
- Branch is generally up to date with main Github - definitely no conflicts
- No unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- PR description says *what* changed and *why*, with a link to the JIRA story.
- Diff has been checked for unexpected changes being included.
- Commit messages say why the change was made.

### CODE REVIEW

File: app/models/calculation_result.rb [hunk 1]
```diff
@@ -285,17 +285,22 @@ def main_home_rows(prefix:)
     {
       value: monetise(capital.dig(:main_home, :value)),
       mortgage: monetise(-capital.dig(:main_home, :outstanding_mortgage)),
-      disregard: monetise(-capital.dig(:main_home, :main_home_equity_disregard)),
-    }
+      disregards: monetise(-capital.dig(:main_home, :main_home_equity_disregard)),
+    }.tap do |ret|
+      transaction_allowance = capital.dig(:main_home, :transaction_allowance).to_d
+      ret[:deductions] = monetise(-transaction_allowance) if transaction_allowance.positive?
+    end
```
Comment (starswan) [inline app/models/calculation_result.rb:292]: would merge be more idiomatic? [Unsure]

File: spec/models/calculation_result_spec.rb [hunk 2]
```diff
@@ -4,4 +4,90 @@
...
+                  additional_properties: [{
+                    value: 0,
+                    outstanding_mortgage: 0,
+                    transaction_allowance: 0,
+                  }],
+                },
+              },
+            },
+          },
+        }
```
Comment (starswan) [inline spec/models/calculation_result_spec.rb:67]: IMHO a FactoryBot helper would remove this noise

File: app/models/calculation_result.rb (hunk 1 above)
Comment (patrick-laa) [reply app/models/calculation_result.rb]: Interestingly, Rubocop complains about a RedundantMerge and tells me to do a simple assignment instead. But `tap` isn't doing much here so I've removed it to make things clearer.

File: spec/models/calculation_result_spec.rb (hunk 2 above)
Comment (patrick-laa) [reply spec/models/calculation_result_spec.rb]: Good idea - I've rebased this PR against EL-712 to give me use of the FactoryBot stuff I added there (although that rebase has slightly screwed up the commit history)

**Review by starswan — APPROVED** [review]

**Pull request merged**